src/SDKs/KeyVault/dataPlane/Microsoft.Azure.KeyVault.TestFramework/KeyVaultTestFixture.cs

```diff
@@ -159,12 +169,39 @@ private bool FromConfiguration()
             return result.AccessToken;
         }
 
+        public async Task<string> GetUserAccessToken(string authority, string resource, string scope)
+        {
+            string clientId = TestConfigurationManager.TryGetEnvironmentOrAppSetting("NativeClientId");
+            var context = new AuthenticationContext(authority, tokenCache);
+            if (_deviceCodeForStorageTests == null)
+            {
+                _deviceCodeForStorageTests =
+                    await context.AcquireDeviceCodeAsync(resource, clientId).ConfigureAwait(false);
+
+                Debug.WriteLine("############################################################################################");
```

Comment:
@dragav, this is not a merge placeholder: it's a write line for output separation.